Add support for TZ in the RTC, used for date formatting

fiam opened the pull request:
A bit more of work towards fixing #102

fiam committed:
Add MSP_RTC message

Returns the RTC from the FC to the sender. Note that MSP_SET_RTC
has been changed to 247 and MSP_RTC is now code 246. This follows
the convention of other MSP message pairs, where the getter has
a lower number than the setter.

Since there are still no MSP senders which send these messages,
there should be no problem with the code modification.

fiam committed:
Add support for configurable TZ offset for the RTC

- Add PG_TIME_CONFIG, defined by the timeConfig_t struct.
- Add tz_offset CLI parameter, which sets the TZ offset in minutes.
- Replace dateTimeFormat() with dateTimeFormatUTC() and
  dateTimeFormatLocal().

fiam committed:
Make cmdMSP uint16_t

uint8_t can't hold MSPv2 commands, so we need to increase its size
in mspFcProcessOutCommand(), mspFcProcessInCommand() and
mspFcProcessCommand().

fiam committed:
Add MSP2_INAV_TZ and MSP2_INAV_SET_TZ MSP commands

MSPv2 only, defined as 0x1001 and 0x1002

martinbudden commented on src/main/common/time.c:
```diff
@@ -58,7 +69,7 @@ static rtcTime_t dateTimeToRtcTime(dateTime_t *dt)
 static void rtcTimeToDateTime(dateTime_t *dt, rtcTime_t *t)
 {
-    int32_t unix = *t / 1000;
+    int32_t unix = *t / MILLIS_PER_SECOND;
```
Can you rename variable `unixTime`?

martinbudden commented on src/main/common/time.c:
```diff
@@ -58,7 +69,7 @@ static rtcTime_t dateTimeToRtcTime(dateTime_t *dt)
 
 static void rtcTimeToDateTime(dateTime_t *dt, rtcTime_t *t)
```
Pass rtcTime by value, ie `rctTime t`

fiam replied:
Should be cheaper to pass the pointer, since rtcTime_t is 64 bits. Not that it would make a significant difference.

martinbudden replied:
With ARM you get up to 4 parameters passed in registers very cheaply. Since `rtcTimeToDateTime` only has two parameters `rtcTime_t` will be passed in two registers. This may even be cheaper than passing by pointer.

martinbudden commented on src/main/common/time.c:
```diff
@@ -86,19 +97,55 @@ static void rtcTimeToDateTime(dateTime_t *dt, rtcTime_t *t)
+    dt->millis = *t % MILLIS_PER_SECOND;
+}
+
+static void dateTimeFormat(char *buf, dateTime_t *dt, int16_t offset)
```
iNav style is not to shorten variable names, so `dateTime_t *dateTime`

martinbudden commented on src/main/common/time.c:
```diff
@@ -24,12 +24,18 @@
  * @author Alberto Garcia Hierro <alberto@garciahierro.com>
  */
 
+#include <stdint.h> // required for common/maths.h
```
If `common/maths.h` needs `stdint.h` then include it there, not here.

fiam committed:
Style changes as requested by @martinbudden

fiam committed:
Add stdint.h include to common/maths.h

maths.h uses types from stdint, so it either needs to include this
header or call the callers need to include it. Also, remove
the stdint.h include from time.c, since it's not needed anymore.

digitalentity commented on src/main/fc/fc_msp.c:
```diff
@@ -532,7 +532,7 @@ static void serializeDataflashReadReply(sbuf_t *dst, uint32_t address, uint16_t
  * May set mspPostProcessFunc to a function to be called once the command has been processed
  */
-static bool mspFcProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst, mspPostProcessFnPtr *mspPostProcessFn)
+static bool mspFcProcessOutCommand(uint16_t cmdMSP, sbuf_t *dst, mspPostProcessFnPtr *mspPostProcessFn)
```
Great! I missed that one when doing MSPv2

digitalentity commented on src/main/msp/msp_protocol_v2_inav.h:
```diff
@@ -15,4 +15,6 @@
-//#define MSP2_INAV_STATUS            0x1000
\ No newline at end of file
+//#define MSP2_INAV_STATUS            0x1000
+#define MSP2_INAV_TZ                0x1001  //out message   Gets the TZ offset for the local time (returns: minutes(i16))
```
I didn't have time to change this file yet, but this API should belong to `msp_protocol_v2_common.h` file and not INAV-specific.

fiam committed:
Rename MSP2_INAV_TZ and MSP2_INAV_SET_TZ

MSP2_COMMON_TZ and MSP2_COMMON_SET_TZ are the new names. Also,
move the definitions to the file msp_protocol_v2_common.h
